docs/rules/no-invalid-debug-function-arguments.md

```diff
@@ -0,0 +1,42 @@
+# Catch usages of Ember&#39;s `assert()` / `warn()` / `deprecate()` functions that have the arguments passed in the wrong order. (no-invalid-debug-function-arguments)
+
+This rule aims to catch a common mistake when using Ember's debug functions:
+* `assert(String description, Boolean condition)`
+* `warn(String description, Boolean condition)`
+* `deprecate(String description, Boolean condition)`
+
+When calling one of these functions, the author may mistakenly pass the arguments in the reverse order, and not notice because the function will be silent with a truthy string as the `condition`.
+
+## Rule Details
+
+Examples of **incorrect** code for this rule:
+
+```js
+import { assert, warn } from '@ember/debug';
+import { deprecate } from '@ember/application/deprecations';
+
+...
+
+assert(label, 'Label must be present.');
+warn(label, 'Label must be present.');
+deprecate(title, 'Title is no longer supported.');
```

Comment:
@bmish I think `deprecate()` is supposed to always have a third argument (an options hash).

Reply:
Good point, added the third argument to both `warn` and `deprecate`.